src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2080,15 +2088,24 @@ TEST(SpatialPoolerTest, testConstructorVsInitialize) {
 }
 
 TEST(SpatialPoolerTest, ExactOutput) { 
-  string gold =
-    "SDR 1 "
-    "1 200 "
-    "10 190 172 23 118 178 129 113 71 185 182 "
-    "~SDR"; // This is all one string.
-
-  stringstream gold_stream( gold );
+  /**** TODO:  fix it.
+  SDR copper_sdr();
+  string copper = "SDR( 200 ) 190, 172, 23, 118, 178, 129, 113, 71, 185, 182\n";
+  std::stringstream copper_stream( copper );
+  copper_stream >> copper_sdr;
+  ****/
+
+  SDR silver_sdr({ 200 });
+  SDR_sparse_t data = {190, 172, 23, 118, 178, 129, 113, 71, 185, 182};
+  silver_sdr.setSparse(data);
+  //silver_sdr.save(std::cout, JSON);
+
+  string gold = "{\"dimensions\": [200],\"sparse\": [190,172,23,118,178,129,113,71,185,182]}";
+  std::stringstream gold_stream( gold );
   SDR gold_sdr;
-  gold_sdr.load( gold_stream );
+  gold_sdr.load( gold_stream, JSON );
+  EXPECT_EQ(silver_sdr, gold_sdr);
```

Comment:
Can you remove two of these SDRs and keep just one?

Reply:
> Can you remove two of these SDRs and keep just one?

I was trying to test two different ways to load an SDR using deserialization.
- silver is loaded directly using initialization and that becomes the standard.
- gold is being loaded via Cereal from a manually written JSON string.
- copper is being loaded via operator>>.  The operator>> should be able to read the output from the operator<< output that you preferred.   

However,  I could not get the operator>> test to work right away.  It was kind of outside the scope of this PR and I did not want to spend a lot of time on it so I commented it out with a TODO.  I will spend a little more time on that.  I think that would be nice-to-have.

Reply:
Yes, it would be nice to have, but not critical.

An optional nitpick is to move this into the SDR unit tests.  My concern is that if/when the Spatial Pooler code changes (and we want to update the gold string) then we will have more strings to update.

Reply:
>  My concern is that if/when the Spatial Pooler code changes 

Well, yes that is a good point.  That is true will any test data put into any test.  

Ok, I will just remove the operator>> function and if it turns out to be needed it can be added with a PR that addresses this issue.  It really is sort of out of scope for this PR anyway.
There was not a strong reason to be making that change.